retriever/lib/install.py

```diff
@@ -53,6 +54,30 @@ def _install(args, use_cache, debug):
                 print(e)
                 if debug:
                     raise
+    elif args['dataset'].startswith('socrata') and (data_sets_scripts is None):
+        socrata_id = args['dataset'].split('-', 1)[1]
+        resource = find_socrata_dataset_by_id(socrata_id)
+
+        if "error" in resource.keys():
+            if resource["datatype"][0] == "map":
+                print("{} because map type datasets are not supported".format(
+                    resource["error"]))
+            else:
+                print("{} because it is of type {} and not tabular".format(
+                    resource["error"], resource["datatype"][1]))
+        elif len(resource.keys()) == 0:
```

Comment:
This should looks to be hanging, may be it should go up. We also do not need to  use == 0, 
>> elif not resourse.keys()